.pipelines/cni/cilium/cilium-overlay-load-test-template.yaml

```diff
@@ -48,38 +49,80 @@ stages:
+                  
+                  
+                  echo "install Cilium ${CILIUM_VERSION_TAG} onto Overlay Cluster"
+                  # Passes Cilium image to daemonset and deployment
+                  envsubst '${CILIUM_VERSION_TAG},${CILIUM_IMAGE_REGISTRY}' < test/integration/manifests/cilium/daemonset.yaml | kubectl apply -f -
+                  envsubst '${CILIUM_VERSION_TAG},${CILIUM_IMAGE_REGISTRY}' < test/integration/manifests/cilium/deployment.yaml | kubectl apply -f -
+                  kubectl get po -owide -A
+
+                  echo "Deploy Azure-CNS"
+                  sudo -E env "PATH=$PATH" make test-integration AZURE_IPAM_VERSION=$(make azure-ipam-version) CNS_VERSION=$(make cns-version) INSTALL_CNS=true INSTALL_OVERLAY=true CNS_IMAGE_REPO=$(CNS_IMAGE_REPO)
+                  kubectl get po -owide -A
+      - ${{if eq(parameters.hubbleEnabled, true)}}:
+        - job: deploy_cilium_components
+          displayName: Deploy Cilium with Hubble
+          steps:
+            - task: AzureCLI@1
+              displayName: "Install Cilium, CNS, and ip-masq-agent"
+              inputs:
+                azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+                scriptLocation: "inlineScript"
+                scriptType: "bash"
+                addSpnToEnvironment: true
+                inlineScript: |
+                  set -ex
+                  az extension add --name aks-preview
+                  make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}-$(commitID)
+                  ls -lah
+                  pwd
+                  kubectl cluster-info
+                  kubectl get po -owide -A
 
-                echo "deploy Cilium ConfigMap"
-                kubectl apply -f test/integration/manifests/cilium/cilium-config.yaml
-                echo "install Cilium onto Overlay Cluster"
-                kubectl apply -f test/integration/manifests/cilium/cilium-agent
-                kubectl apply -f test/integration/manifests/cilium/cilium-operator
-                echo "install Cilium ${CILIUM_VERSION_TAG} onto Overlay Cluster"
-                # Passes Cilium image to daemonset and deployment
-                envsubst '${CILIUM_VERSION_TAG},${CILIUM_IMAGE_REGISTRY}' < test/integration/manifests/cilium/daemonset.yaml | kubectl apply -f -
-                envsubst '${CILIUM_VERSION_TAG},${CILIUM_IMAGE_REGISTRY}' < test/integration/manifests/cilium/deployment.yaml | kubectl apply -f -
-                kubectl get po -owide -A
+                  echo "deploy Cilium ConfigMap"
+                  kubectl apply -f test/integration/manifests/cilium/v${CILIUM_HUBBLE_VERSION_TAG}/cilium-config/cilium-config-hubble.yaml
+                  
+                  echo "install Cilium onto Overlay Cluster with hubble enabled"
+                  kubectl apply -f test/integration/manifests/cilium/v1.14.4/cilium-agent/files

```

Comment:
Can we parameterize this Or read from cilium agent ? We expect this to change in the future with new versions ?

Reply:
We're keeping the versioned file structure such that the templates will be separate for 1.14.x and 1.15.x, and so on. I am working on a PR to parameterize the versions so that we can still reuse the templates for each patch version. The `cilium/v1.14.4` directory will become `cilium/v1.14`, and we'll parameterize the input in the pipelines so we can use the templates when we take the next patch after 1.14.4. 
https://github.com/Azure/azure-container-networking/pull/2562